bdns/index.js

```diff
@@ -34,21 +34,22 @@ function BDNS() {
                 return callback(Error(`The provided domain ${dlDomain} is not configured`));
             }
 
-            return bdnsCache[dlDomain][section] ? bdnsCache[dlDomain][section] : [getBaseURL()];
+            const config = bdnsCache[dlDomain][section] ? bdnsCache[dlDomain][section] : [getBaseURL()];
+            callback(undefined, config);
         }
 
         if (!isInitialized) {
             return this.addPendingCall(() => {
                 if (dlDomain === undefined) {
                     return callback(new Error("The domain is not defined"));
                 }
-                callback(undefined, load_or_default());
```

Comment:
If a domain is undefined or not configured, the `callback` will be called with (undefined, undefined) causing an exception in `bricking.putBrick()`